### DISCUSSION TITLE
Hyperband without parallel processes

### DISCUSSION THREAD

**Discussion by L-Reichardt** [optuna/optuna#4046] [Q&A] [answered]
Hello, I have a questions about how Hyperband is implemented in Optuna.

Assuming resources_min = 10, and num_runs = 50 its my understanding that hyperband trains all trials to 10 epochs. Then, some trials are selected to be trained an additional 10 epochs.

For "resuming" promising trials, it would be necessary to reload the model and statistics of one trial. How is this implemented in Optuna? Or is the training for these trials started fresh?

Best Regards,
L

**Comment by not522** [comment, marked as answer]
Optuna's Hyperband differs from the original algorithm in continuing rather than resuming training. See Algorithm 1 of Optuna's paper for details. https://arxiv.org/abs/1907.10902

**Reply by L-Reichardt** [reply]
thank you
